jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -240,11 +240,11 @@
 	private static final class MetadataCache {
 		MetadataCache() {}
 
-		private static long cachedCanonicalNameOffset = -1;
-		private static long cachedSimpleNameOffset = -1;
+		static long cachedCanonicalNameOffset;
```

Comment:
Won't removing the initialization to -1 break the code by returning 0 for the offset?
```
		MetadataCache.cachedSimpleNameOffset = getFieldOffset(
				MetadataCache.class, "cachedSimpleName", MetadataCache.cachedSimpleNameOffset);
```
```
private long getFieldOffset(Class<?> hostClass, String fieldName, long initialOffset) {
	if (initialOffset == -1) {
...
	}
	return initialOffset;
```

Reply:
Yes, you're right. I'll restore that initialization.